brave/src/main/java/brave/propagation/CurrentTraceContext.java

```diff
@@ -45,4 +48,59 @@
       return () -> local.set(previous);
     }
   }
-}
\ No newline at end of file
+
+  /** Wraps the input so that it executes with the same context as now. */
+  public <C> Callable<C> wrap(Callable<C> task) {
+    final TraceContext invocationContext = get();
+    return () -> {
+      try (Scope scope = newScope(invocationContext)) {
+        return task.call();
+      }
+    };
+  }
+
+  /** Wraps the input so that it executes with the same context as now. */
+  public Runnable wrap(Runnable task) {
+    final TraceContext invocationContext = get();
+    return () -> {
+      try (Scope scope = newScope(invocationContext)) {
+        task.run();
+      }
+    };
+  }
+
+  /**
+   * Decorates the input such that the {@link #get() current trace context} at the time a task is
+   * scheduled is made current when the task is executed.
+   */
+  public Executor executor(Executor delegate) {
+    class CurrentTraceContextExecutor implements Executor {
+      @Override public void execute(Runnable task) {
+        delegate.execute(CurrentTraceContext.this.wrap(task));
+      }
+    }
+    return new CurrentTraceContextExecutor();
+  }
+
+  /**
+   * Decorates the input such that the {@link #get() current trace context} at the time a task is
+   * scheduled is made current when the task is executed.
+   */
+  public ExecutorService executorService(ExecutorService delegate) {
+    class CurrentTraceContextExecutorService extends brave.internal.WrappingExecutorService {
```

Comment:
doing like this allows extenders to not rely on any brave internal classes if they have their own executorservice